lib/compat/wordpress-6.1/class-wp-theme-json-6-1.php

```diff
@@ -1308,14 +1308,14 @@ protected function get_layout_styles( $block_metadata ) {
 					$block_gap_value = _wp_array_get( $block_type->supports, array( 'spacing', 'blockGap', '__experimentalDefault' ), null );
 				}
 			} else {
-				$block_gap_value = _wp_array_get( $node, array( 'spacing', 'blockGap' ), null );
+				$block_gap_value = static::get_property_value( $node, array( 'spacing', 'blockGap' ) );
```

Comment:
I think we need this also in `get_styles_for_block` [here](https://github.com/WordPress/gutenberg/blob/2d5ff2fb2a263ac8cadd95df693523c12b766487/lib/compat/wordpress-6.1/class-wp-theme-json-6-1.php#L814)

Without it the body CSS rule with custom var isn't parsed:

```
body {
    --wp--style--block-gap: var:preset|spacing|60;
}
```

Reply:
that will be what is breaking the gallery as it relies on `--wp--style--block-gap` as the fallback gap value if no block level gap is set

Reply:
Nice one, thanks for catching that! I'll update 👍

Good eyes! 🦅 👀

Reply:
I've updated in https://github.com/WordPress/gutenberg/pull/43296/commits/2f1d6876168511db7055677508e813ae3c1f9e28, and the Gallery block appears to be working now. I also removed the `0.5em` fallback in that case, because it'll never be reached (the rule is only output if block gap is enabled, in which case, there will always be a real blockGap value because the root theme.json provides one)

It looks like the site editor was already parsing the value for the JS side of things, but let me know if you encounter any other issues~!

Reply:
A question: should the following block style in theme.json have any effect?

```json
			"core/gallery": {
				"spacing": {
					"blockGap": "var:preset|spacing|80"
				}
			}
```